src/layout/node.ts

```diff
@@ -20,17 +22,23 @@ export default function(node: Node, source: Source): void {
         node = node.parentNode;
     }
 
-    let call = dom.has(node) ? "update" : "add";
+    let add = dom.has(node) === false;
+    let call = add ? "add" : "update";
     let parent = node.parentElement ? node.parentElement : null;
+    let insideFrame = node.ownerDocument !== document;
     switch (node.nodeType) {
-        case Node.DOCUMENT_NODE:
-            observe(node);
-            break;
         case Node.DOCUMENT_TYPE_NODE:
+            // Skip processing nodes within iframe if they were already removed from the DOM before we got to processing them.
```

Comment:
How could this happen?

Reply:
Simplified this. Now handle it during rendering. For reference the way we would get into this state is: our code is async so by the time we are able to get to iframe, it's possible it may already have been destroyed.